### PR TITLE
Added test for gmm_generate

### PR DESCRIPTION
**Pull request by gauravcr7rm**
Added test for gmm_generate

### CODE REVIEW

File: src/mlpack/tests/main_tests/gmm_generate_test.cpp [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+BOOST_FIXTURE_TEST_SUITE(GmmGenerateMainTest,
+                         GmmGenerateTestFixture);
+
+// Making sure samples are provided.
+BOOST_AUTO_TEST_CASE(GmmGenerateSamples)
+{
+  arma::mat inputData;
+  if (!data::Load("vc2.csv", inputData))
+  BOOST_FAIL("Unable to load train dataset vc2.csv!");
```
Comment (rcurtin) [inline src/mlpack/tests/main_tests/gmm_generate_test.cpp]: This should be indented.

File: src/mlpack/tests/main_tests/gmm_generate_test.cpp [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+  SetInputParam("input_model", std::move(gmm));
+
+  Log::Fatal.ignoreInput = true;
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+}
+
+// Checking dimensionality of output.
+BOOST_AUTO_TEST_CASE(GmmGenerateDimensionality)
+{ int samples =10;
```
Comment (rcurtin) [inline src/mlpack/tests/main_tests/gmm_generate_test.cpp]: Why is this on the same line as the opening brace?

File: src/mlpack/tests/main_tests/gmm_generate_test.cpp [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+  
+  GMM gmm(1, 2);
+  gmm.Train(data, 10);
+
+  SetInputParam("input_model", std::move(gmm));
+  SetInputParam("samples", samples);
+
+  mlpackMain();
+
+  arma::mat output = CLI::GetParam<arma::mat>("output");
```
Comment (rcurtin) [inline src/mlpack/tests/main_tests/gmm_generate_test.cpp]: No need for a copy here; you can use a reference like `const arma::mat& output = ...`, or move the matrix with `arma::mat output = std::move(CLI::GetParam<arma::mat>("output"));`.

File: src/mlpack/tests/main_tests/gmm_generate_test.cpp [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+  BOOST_REQUIRE_EQUAL(output.n_rows, gmm->Dimensionality());
+  BOOST_REQUIRE_EQUAL(output.n_cols, samples);
+ 
+}
+
+// Samples passed should be positive.
```
Comment (rcurtin) [inline src/mlpack/tests/main_tests/gmm_generate_test.cpp]: There is a related test, which is to see what happens when the number of samples passed is 0.

Comment (gauravcr7rm) [reply src/mlpack/tests/main_tests/gmm_generate_test.cpp]: ok ill do the changes

Comment (gauravcr7rm) [reply src/mlpack/tests/main_tests/gmm_generate_test.cpp]: @rcurtin i was able to build all these successfully in my local system and  i have made all the changes specified.
Please have a look

File: src/mlpack/tests/main_tests/gmm_generate_test.cpp [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+// Making sure samples are provided.
+BOOST_AUTO_TEST_CASE(GmmGenerateSamples)
+{
+  arma::mat inputData;
+  if (!data::Load("vc2.csv", inputData))
+    BOOST_FAIL("Unable to load train dataset vc2.csv!");
+  
+  GMM gmm(1, 2);
+  gmm.Train(inputData, 2);
+  SetInputParam("input_model", std::move(gmm));
```
Comment (rcurtin) [inline src/mlpack/tests/main_tests/gmm_generate_test.cpp]: I think you need to call this with a pointer to the model, so `SetInputParam("input_model", &gmm);`.  If you fix that, I believe the build will pass.

**Comment by rcurtin** [comment]
With #1899 merged this is handled so I'll close this.  Thanks again for the contribution. :+1:

**Pull request closed without merging**